org.eclipse.debug.core/core/org/eclipse/debug/internal/core/LaunchManager.java

```diff
@@ -530,20 +530,27 @@ public static Document getDocument() throws ParserConfigurationException {
 	 * platform line separators.
 	 *
 	 * @param doc document to serialize
+	 * @param lineDelimiter the new line separator to use
 	 * @return the document as a string
 	 * @throws TransformerException if an unrecoverable error occurs during the
 	 *             serialization
 	 * @throws IOException if I/O error occurs
 	 */
-	public static String serializeDocument(Document doc) throws TransformerException, IOException {
+	public static String serializeDocument(Document doc, String lineDelimiter) throws TransformerException, IOException {
 		ByteArrayOutputStream s = new ByteArrayOutputStream();
-		TransformerFactory factory = TransformerFactory.newInstance();
-		Transformer transformer = factory.newTransformer();
-		transformer.setOutputProperty(OutputKeys.METHOD, "xml"); //$NON-NLS-1$
-		transformer.setOutputProperty(OutputKeys.INDENT, "yes"); //$NON-NLS-1$
-		DOMSource source = new DOMSource(doc);
-		StreamResult outputTarget = new StreamResult(s);
-		transformer.transform(source, outputTarget);
+		try {
+			final DOMImplementationLS domRegistry = (DOMImplementationLS) DOMImplementationRegistry.newInstance().getDOMImplementation("LS"); //$NON-NLS-1$
+			final LSSerializer serializer = domRegistry.createLSSerializer();
+			serializer.getDomConfig().setParameter("format-pretty-print", true); //$NON-NLS-1$
+			serializer.getDomConfig().setParameter("http://www.oracle.com/xml/jaxp/properties/isStandalone", true); //$NON-NLS-1$
+			serializer.setNewLine(lineDelimiter);
+			final LSOutput destination = domRegistry.createLSOutput();
+			destination.setEncoding(StandardCharsets.UTF_8.name());
```

Comment:
why utf8?

Reply:
please change the copyright year in the changed files to 2022

Reply:
Done the copyright changes.

The UTF-8 is because that is what was originally in place in the XML header and what the previous transformer did.

I guess this could be changed as well to honor workspace settings, but that is not what people are complaining about in the bugs.The complaint people have is that they add the launch files to version control and they show up as modified files when cloned across Windows/Unix machines.

If I were to change this line, I'd need to update the XML header as well to `<?xml version="1.0" encoding="__whatever__?>`. But I am not sure this should be mixed with this change.